Remove protocol 1 and 2

bigmontz opened the pull request:
The reference this protocols are not needed since it's not used by the driver anymore.

During the remotion of these protocol version, the tests were refactoried to be specific per version instead of
assume the re-use done on the protocols implementation.

gjmwoods commented on driver/src/test/java/org/neo4j/driver/internal/InternalResultTest.java:
```diff
@@ -19,46 +19,42 @@
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.*;
```
Could you double-check the neo4j checkstyles is being applied? Pretty sure its doesn't use *. (Personally not fussed either way just makes reviews easier)

bigmontz replied:
I will check again.

bigmontz replied:
It is, interesting it is not being applied correctly.

bigmontz replied:
I've reseted the commit, and then applied the reformatting on all the modified files.

gjmwoods commented on driver/src/test/java/org/neo4j/driver/internal/messaging/v42/MessageFormatV42Test.java:
```diff
@@ -0,0 +1,61 @@
+ */
+class MessageFormatV42Test
+{
+    private static final MessageFormat format = BoltProtocolV4.INSTANCE.createMessageFormat();
```
`BoltProtocolV42`?

bigmontz replied:
Yes, I will fix it.

gjmwoods reviewed (commented):
Awesome stuff! Just double-check the check styles setup before merging.

bigmontz committed:
Remove protocol 1 and 2

The reference this protocols are not needed since it's not used by the driver anymore.

During the remotion of these protocol version, the tests were refactoried to be specific per version instead of
assume the re-use done on the protocols implementation.
